python/src/nanoarrow/c_lib.py

```diff
@@ -125,10 +138,205 @@ def c_array(obj=None, requested_schema=None) -> CArray:
         out = CArray.allocate(CSchema.allocate())
         obj._export_to_c(out._addr(), out.schema._addr())
         return out
-    else:
+
+    # Try buffer protocol (e.g., numpy arrays)
+    try:
+        return c_array_from_pybuffer(obj)
+    except Exception as e:
         raise TypeError(
             f"Can't convert object of type {type(obj).__name__} to nanoarrow.c_array"
+        ) from e
+
+
+def c_array_from_pybuffer(obj) -> CArray:
```

Comment:
(Optional) It might be nice to put these into a class e.g.
```
class CArray:
    @staticmethod
    def from_pybuffers():
         ...

    @staticmethod
    def from_buffers():
         ...